addons/pos_self_order/models/pos_config.py

```diff
@@ -291,8 +292,7 @@ def _load_pos_self_data_read(self, records, config):
         record['_self_ordering_image_background_ids'] = config.self_ordering_image_background_ids.ids
         record['_pos_special_products_ids'] = config._get_special_products().ids
         record['_self_ordering_style'] = {
-            'primaryBgColor': self.env.company.email_secondary_color,
-            'primaryTextColor': self.env.company.email_primary_color,
+            'primaryBgColor': config.self_ordering_primary_color,
```

Comment:
Since its a field directly on PoS config you don't need to load it like that, it will be directly accessible in the JS via `config.self_ordering_primary_color`